system/src/system_power_manager.cpp

```diff
@@ -123,28 +123,29 @@ void PowerManager::update() {
 }
 
 void PowerManager::sleep(bool s) {
+  if (isRunning()) {
 #if HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL
```

Comment:
Nit: formatting.

This can also probably be changed to:
```c++
#if HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL
if (detect_ && isRunning()) {
#else
if (isRunning()) {
#endif //HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL
```